compiler/typecheck/TcExpr.hs

```diff
@@ -1845,7 +1859,7 @@ tc_infer_id lbl id_name
                  theta' = substTheta subst theta
                  rho'   = substTy subst rho
            ; wrap <- instCall (OccurrenceOf id_name) tys' theta'
-           ; addDataConStupidTheta con tys'
+           ; addDataConStupidTheta con (tail tys') -- tail for extra multiplicity argument
```

Comment:
This is not a very useful/understandable comment.

Something like

> We use 'tail' to skip the multiplicity argument. We don't want the multiplicity argument because…